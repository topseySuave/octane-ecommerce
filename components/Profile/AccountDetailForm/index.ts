import AccountDetailForm from './AccountDetailForm';
export default AccountDetailForm;
