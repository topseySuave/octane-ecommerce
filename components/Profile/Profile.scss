.oct-profile-container {
  padding: 50px;
  display: flex;
  justify-content: center;
  align-content: center;
}