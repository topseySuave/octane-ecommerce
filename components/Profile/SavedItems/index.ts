import SavedItems from './SavedItems';

export default SavedItems;
