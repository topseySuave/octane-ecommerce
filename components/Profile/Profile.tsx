import { Button, Card, Layout, Tabs, Typography } from 'antd';
import { DISTANCE_FROM_TOP, LINE_HEIGHT, isWindows } from 'lib/constants';
import Link from 'next/link';
import AccountDetailForm from './AccountDetailForm';
import Orders from './Orders';
import './Profile.scss';
import SavedItems from './SavedItems';
import ShippingForm from './ShippingForm';
import { useEffect } from 'react';
import { getUserData } from 'lib/utils';
import Router from 'next/router';

const { Title } = Typography;
const { Content } = Layout;
const { TabPane } = Tabs;

const Profile = () => {
  useEffect(() => {
    if (isWindows && !getUserData().accessToken) Router.push('/shop');
  }, []);

  return (
    <Content style={{ marginTop: DISTANCE_FROM_TOP + LINE_HEIGHT }}>
      <div className="oct-profile-container">
        <Title level={2}>Account Details</Title>
        <Card style={{ width: '100%' }}>
          <Tabs tabPosition={'left'}>
            <TabPane tab="My Octane Account" key="1">
              <AccountDetailForm />
            </TabPane>
            <TabPane tab="Account/Shipping Address" key="2">
              <ShippingForm />
            </TabPane>
            <TabPane tab="Orders" key="3">
              <Orders />
            </TabPane>
            <TabPane tab="Saved for Later" key="4">
              <SavedItems />
            </TabPane>
          </Tabs>
          <Link prefetch href="/signout">
            <a>
              <Button type="link">Log out</Button>
            </a>
          </Link>
        </Card>
      </div>
    </Content>
  );
};

export default Profile;
