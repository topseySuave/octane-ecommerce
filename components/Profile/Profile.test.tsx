import { shallow } from 'enzyme';
import React from 'react';

import Profile from './Profile';

describe('Profile', () => {
  it('Renders with user profile tabs settings', () => {
    const tree = shallow(<Profile />);
    expect(tree).toMatchSnapshot();
    // const mounted = shallow(<Profile />);
    // const profile = mounted.find('content');
    // console.log(profile);
  });
});
