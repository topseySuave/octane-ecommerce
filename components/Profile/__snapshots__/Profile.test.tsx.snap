// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Renders with user profile tabs settings 1`] = `ShallowWrapper {}`;
